File: benchmarks/tests/loop_intensive.py
#!/usr/bin/env python3
"""
Python 循环密集型性能测试
测试各种循环的执行效率
"""

def nested_loops(n):
    """嵌套循环测试"""
    sum_val = 0
    i = 1
    
    while i <= n:
        j = 1
        while j <= n:
            sum_val += i * j
            j += 1
        i += 1
    
    return sum_val

def for_loop_sum(n):
    """for循环求和"""
    sum_val = 0
    
    for i in range(1, n + 1):
        sum_val += i
    
    return sum_val

def while_loop_sum(n):
    """while循环求和"""
    sum_val = 0
    i = 1
    
    while i <= n:
        sum_val += i
        i += 1
    
    return sum_val

def array_iteration(size):
    """数组迭代测试"""
    # 创建数组
    arr = []
    i = 0
    
    # 填充数组
    while i < size:
        arr.append(i)
        i += 1
    
    # 计算数组元素和
    sum_val = 0
    j = 0
    while j < len(arr):
        sum_val += arr[j]
        j += 1
    
    return sum_val

def complex_loop_calculation(n):
    """复杂循环计算"""
    result = 0
    
    for i in range(1, n + 1):
        temp = 0
        j = 1
        
        while j <= i:
            if j % 2 == 0:
                temp += j * j
            else:
                temp += j
            j += 1
        
        result += temp
    
    return result

def main():
    print("=== Python 循环密集型性能测试 ===")
    
    # 测试参数
    n = 500
    array_size = 1000
    
    print("测试参数:")
    print(f"n = {n}")
    print(f"array_size = {array_size}")
    print()
    
    # 1. 嵌套循环测试
    print("1. 嵌套循环测试:")
    nested_result = nested_loops(n)
    print(f"nested_loops({n}) = {nested_result}")
    print()
    
    # 2. for循环测试
    print("2. for循环测试:")
    for_result = for_loop_sum(n)
    print(f"for_loop_sum({n}) = {for_result}")
    print()
    
    # 3. while循环测试
    print("3. while循环测试:")
    while_result = while_loop_sum(n)
    print(f"while_loop_sum({n}) = {while_result}")
    print()
    
    # 验证for和while循环结果一致
    if for_result == while_result:
        print("✓ for和while循环结果一致")
    else:
        print("✗ for和while循环结果不一致")
    print()
    
    # 4. 数组迭代测试
    print("4. 数组迭代测试:")
    array_result = array_iteration(array_size)
    print(f"array_iteration({array_size}) = {array_result}")
    print()
    
    # 5. 复杂循环计算测试
    print("5. 复杂循环计算测试:")
    complex_result = complex_loop_calculation(n)
    print(f"complex_loop_calculation({n}) = {complex_result}")
    print()
    
    print("=== 循环密集型测试完成 ===")

if __name__ == "__main__":
    main()

File: others/benchmarks/tests/loop_intensive.php
<?php
/**
 * PHP 循环密集型性能测试
 * 对应 CodeNothing v0.5.11 循环优化测试
 */

function nestedLoops($n) {
    $sum = 0;
    $i = 1;
    
    while ($i <= $n) {
        $j = 1;
        while ($j <= $n) {
            $sum = $sum + ($i * $j);
            $j = $j + 1;
        }
        $i = $i + 1;
    }
    
    return $sum;
}

function forLoopSum($n) {
    $sum = 0;
    
    for ($i = 1; $i <= $n; $i++) {
        $sum = $sum + $i;
    }
    
    return $sum;
}

function whileLoopSum($n) {
    $sum = 0;
    $i = 1;
    
    while ($i <= $n) {
        $sum = $sum + $i;
        $i = $i + 1;
    }
    
    return $sum;
}

function arrayIteration($size) {
    // 简化版本：直接计算1到size的和
    $sum = 0;
    $i = 0;
    
    while ($i < $size) {
        $sum = $sum + $i;
        $i = $i + 1;
    }
    
    return $sum;
}

function complexLoopCalculation($n) {
    $result = 0;
    
    for ($i = 1; $i <= $n; $i++) {
        $temp = 0;
        $j = 1;
        
        while ($j <= $i) {
            if ($j % 2 == 0) {
                $temp = $temp + $j * $j;
            } else {
                $temp = $temp + $j;
            }
            $j = $j + 1;
        }
        
        $result = $result + $temp;
    }
    
    return $result;
}

function main() {
    echo "=== PHP 循环密集型性能测试 ===\n";
    
    // 测试参数
    $n = 100;
    $arraySize = 200;
    
    echo "测试参数:\n";
    echo "n = " . $n . "\n";
    echo "arraySize = " . $arraySize . "\n";
    echo "\n";
    
    // 1. 嵌套循环测试
    echo "1. 嵌套循环测试:\n";
    $nestedResult = nestedLoops($n);
    echo "nestedLoops(" . $n . ") = " . $nestedResult . "\n";
    echo "\n";
    
    // 2. for循环测试
    echo "2. for循环测试:\n";
    $forResult = forLoopSum($n);
    echo "forLoopSum(" . $n . ") = " . $forResult . "\n";
    echo "\n";
    
    // 3. while循环测试
    echo "3. while循环测试:\n";
    $whileResult = whileLoopSum($n);
    echo "whileLoopSum(" . $n . ") = " . $whileResult . "\n";
    echo "\n";
    
    // 验证for和while循环结果一致
    if ($forResult == $whileResult) {
        echo "✓ for和while循环结果一致\n";
    } else {
        echo "✗ for和while循环结果不一致\n";
    }
    echo "\n";
    
    // 4. 数组迭代测试
    echo "4. 数组迭代测试:\n";
    $arrayResult = arrayIteration($arraySize);
    echo "arrayIteration(" . $arraySize . ") = " . $arrayResult . "\n";
    echo "\n";
    
    // 5. 复杂循环计算测试
    echo "5. 复杂循环计算测试:\n";
    $complexResult = complexLoopCalculation($n);
    echo "complexLoopCalculation(" . $n . ") = " . $complexResult . "\n";
    echo "\n";
    
    echo "=== 循环密集型测试完成 ===\n";
    
    return 0;
}

// 执行主函数
main();
?>
